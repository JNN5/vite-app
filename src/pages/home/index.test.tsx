import { screen } from "@testing-library/react";
import { render } from "../../test-setup/test-utils";
import userEvent from "@testing-library/user-event";

import Page from ".";

describe("Page", () => {
  it("renders page", () => {
    render(<Page />);

    screen.debug();
  });

  it("matches screenshot", () => {
    const { asFragment } = render(<Page />);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Interactions", () => {
  it("counter adds 3", async () => {
    // Given
    render(<Page />);

    // When
    await userEvent.click(screen.getByText("count is 0"));

    // Then
    expect(screen.getByText("count is 3")).toBeVisible();
  });
});
