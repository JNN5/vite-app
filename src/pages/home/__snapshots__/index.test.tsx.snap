// Vitest Snapshot v1

exports[`Page > matches screenshot 1`] = `
<DocumentFragment>
  <div>
    <a
      href="https://vitejs.dev"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="/vite.svg"
      />
    </a>
    <a
      href="https://reactjs.org"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button>
      count is 0
    </button>
    <div>
      <p>
        Dogs to fetch:
      </p>
      <select>
        <option
          value="5"
        >
          5
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="20"
        >
          20
        </option>
      </select>
    </div>
    <div>
      <p>
        Number of dogs fetched: 0
      </p>
      <table>
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Picture
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</DocumentFragment>
`;
